acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/publickeys/PublicKeysAcceptanceTestBase.java

```diff
@@ -28,52 +29,117 @@
   static final String SIGNER_PUBLIC_KEYS_PATH = "/signer/publicKeys";
+  static final String BLS = "BLS";
+  static final String SECP256K1 = "SECP256K1";
 
-  private static final String PRIVATE_KEY_1 =
+  private static final String BLS_PRIVATE_KEY_1 =
       "3ee2224386c82ffea477e2adf28a2929f5c349165a4196158c7f3a2ecca40f35";
-  private static final String PRIVATE_KEY_2 =
+  private static final String BLS_PRIVATE_KEY_2 =
       "32ae313afff2daa2ef7005a7f834bdf291855608fe82c24d30be6ac2017093a8";
+  private static final String SECP_PRIVATE_KEY_1 =
+      "d392469474ec227b9ec4be232b402a0490045478ab621ca559d166965f0ffd32";
+  private static final String SECP_PRIVATE_KEY_2 =
+      "2e322a5f72c525422dc275e006d5cb3954ca5e02e9610fae0ed4cc389f622f33";
 
   private static final MetadataFileHelpers metadataFileHelpers = new MetadataFileHelpers();
 
   @TempDir Path testDirectory;
 
-  protected String[] privateKeys() {
-    return new String[] {PRIVATE_KEY_1, PRIVATE_KEY_2};
+  protected String[] privateKeys(final String keyType) {
+    return keyType.equals("BLS")
+        ? new String[] {BLS_PRIVATE_KEY_1, BLS_PRIVATE_KEY_2}
+        : new String[] {SECP_PRIVATE_KEY_1, SECP_PRIVATE_KEY_2};
   }
 
-  protected String[] createKeys(boolean isValid, final String... privateKeys) {
+  protected String[] createKeys(
+      final String keyType, boolean isValid, final String... privateKeys) {
+    return keyType.equals("BLS")
+        ? createBlsKeys(isValid, privateKeys)
+        : createSecpKeys(isValid, privateKeys);
+  }
+
+  protected String[] createBlsKeys(boolean isValid, final String... privateKeys) {
     return Stream.of(privateKeys)
         .map(
             privateKey -> {
               final BLSKeyPair keyPair =
                   new BLSKeyPair(BLSSecretKey.fromBytes(Bytes.fromHexString(privateKey)));
-              final Path keyConfigFile = configFileName(keyPair.getPublicKey());
+              final Path keyConfigFile = blsConfigFileName(keyPair.getPublicKey());
               if (isValid) {
                 metadataFileHelpers.createUnencryptedYamlFileAt(keyConfigFile, privateKey);
               } else {
-                try {
-                  Files.createFile(keyConfigFile);
-                } catch (final IOException e) {
-                  throw new UncheckedIOException(e);
-                }
+                createInvalidFile(keyConfigFile);
               }
               return keyPair.getPublicKey().toString();
             })
         .toArray(String[]::new);
   }
 
-  private Path configFileName(final BLSPublicKey publicKey) {
-    final String configFilename2 = publicKey.toString().substring(2);
-    return testDirectory.resolve(configFilename2 + ".yaml");
+  protected String[] createSecpKeys(boolean isValid, final String... privateKeys) {
+    return Stream.of(privateKeys)
+        .map(
+            privateKey -> {
+              final ECKeyPair ecKeyPair =
```

Comment:
is there not a cleaner way to get to the Public Key without going via ECKeyPair?

Reply:
don't think so